content/guides/getstarted/solana-test-validator.md

```diff
@@ -0,0 +1,342 @@
+```
+
+## Changing Versions
+
+To check your current `solana-test-validator` version:
+
+```shell
+solana-test-validator --version
+```
+
+Your `solana-test-validator` runs on the same version as the Solana CLI version.
+
+To test your programs against different versions of the Solana runtime, you can
+install multiple versions of the Solana CLI and switch between them using the
+solana-install set command:
+
+```shell
+solana-install init <VERSION>
+```
+
+- `VERSION` is the desired CLI version to install
+
+Make sure to restart your Solana test validator after changing versions to
+ensure it runs the correct version.
+
+## Cloning Programs
+
+To add existing on-chain programs to your local environment, you can clone the
+program.
+
+To clone an account from the cluster:
+
+```shell
+solana-test-validator --clone PROGRAM_ADDRESS
+```
+
+This is useful for testing interactions with standard programs.
+
+To clone an upgradeable program and its executable data from the cluster:
+
+```shell
+solana-test-validator --clone-upgradeable-program PROGRAM_ADDRESS
+```
+
+## Resetting State on Accounts at Startup
+
+By default the validator will resume an existing ledger _(if present)_. But
+during startup, you can reset the ledger either to genesis or to specific
+account state that you provide.
+
+### Reset to Genesis
+
+To reset the ledger to the genesis state:
+
+```shell
+solana-test-validator --reset
+```
+
+### Reset to Specific Accounts
+
+To reset the state of specific accounts every time you start the validator, you
+can use a combination of account snapshots and the `--account` flag.
+
+First, save the desired state of an account as a JSON file:
+
+```shell
+solana account PROGRAM_ADDRESS --output json > account_state.json
+```
+
+Then load this state each time you reset the validator:
+
+```shell
+solana-test-validator --reset --account PROGRAM_ADDRESS account_state.json
+```
+
+## Interacting with a Running Test Validator
+
+Once you have the `solana-test-validator` up and running, you can interact with
+it using various Solana CLI (Command Line Interface) commands. These commands
+let you [deploy programs](/docs/programs/deploying.md), manage
+[accounts](/docs/core/accounts.md), send
+[transactions](/docs/core/transactions.md), and much more. Here’s a detailed
+guide on the key commands you will use.
+
+### Checking the Status of the Test Validator
+
+Before interacting with the test validator, it's useful to confirm its status
+and ensure it is running correctly.
+
+```shell
+solana ping
+```
+
+This command pings the local test validator and returns the current blockhash
+and latency, verifying that it is active.
+
+### Account Management
+
+To create a new keypair (account), use:
+
+```shell
+solana-keygen new
+```
+
+This command creates a new keypair and saves it to the specified file.
+
+To retrieve details about an account, such as its balance and owner:
+
+```shell
+solana account <ACCOUNT_ADDRESS>
```

Comment:
If you do airdrop before `solana account X` you can view the details of the account. If you do before, the account will not exist.